src/kakarot/library.cairo

```diff
@@ -269,9 +282,118 @@ namespace Kakarot {
         pedersen_ptr: HashBuiltin*,
         range_check_ptr,
         bitwise_ptr: BitwiseBuiltin*,
-    }(evm_address: felt) -> (starknet_contract_address: felt) {
+    }(evm_contract_address: felt) -> (starknet_contract_address: felt) {
         let (class_hash) = externally_owned_account_class_hash.read();
-        let (starknet_contract_address) = Accounts.create(class_hash, evm_address);
+        let (starknet_contract_address) = Accounts.create(class_hash, evm_contract_address);
         return (starknet_contract_address=starknet_contract_address);
     }
+
+    // @notice The eth_call function as described in the RPC spec, see https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_call
+    // @dev "from" parameter is taken from the get_caller_address syscall
+    // @param to The address the transaction is directed to.
+    // @param gas_limit Integer of the gas provided for the transaction execution
+    // @param gas_price Integer of the gas price used for each paid gas
+    // @param value Integer of the value sent with this transaction
+    // @param data_len The length of the data
+    // @param data Hash of the method signature and encoded parameters. For details see Ethereum Contract ABI in the Solidity documentation
+    // @return return_data_len The length of the returned bytes
+    // @return return_data The returned bytes array
+    func eth_call{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr,
+        bitwise_ptr: BitwiseBuiltin*,
+    }(to: felt, gas_limit: felt, gas_price: felt, value: felt, data_len: felt, data: felt*) -> (
+        return_data_len: felt, return_data: felt*
+    ) {
+        let (success) = transfer(to, value);
```

Comment:
I don't think eth_call is allowed to transfer value, as there is no transaction posted on the block chain
![image](https://user-images.githubusercontent.com/66871571/228813887-272bf447-d59b-4c30-b9da-6e78d38da4da.png)

Reply:
actually the `eth_call` does have a `value`, "Integer of the value sent with this transaction", and I think that the comment in the doc regarding the gas: eth_call consumes zero gas, but this parameter may be needed by some executions applies to value as well. Even though this is only "view", the value may be required. 

if we consider the tagline of eth_call and eth_sendTransaction
- Executes a new message call immediately without creating a transaction on the block chain.
- Creates new message call transaction or a contract creation, if the data field contains code.

it makes sense to me that they do exactly the same *but* actually writing on the blockchain